base58.go

```diff
@@ -45,16 +51,20 @@ func Base58Decode(b, alphabet string) ([]byte, error) {
 	copy(val[numZeros:], tmpval)
 
 	// Check checksum
-	checksum := doubleSha256(val[0 : len(val)-4])
-	expected := val[len(val)-4:]
+	checksum := doubleSha256(val[: len(val) - 4])
+	expected := val[len(val) - 4:]
 	if !bytes.Equal(checksum[0:4], expected) {
 		return nil, fmt.Errorf("Bad Base58 checksum: %v expected %v", checksum, expected)
 	}
-	return val, nil
+	// strip checksum
+	return val[:len(val) - 4], nil
```

Comment:
can you revert this change? let caller to remove the checksum, we may add hasher to this func to support checksum other than `doubleSha256`

Reply:
Well, I disagree.  Either just to base58 decode, or do checksum change and remove the checksum.  I refactor in Base58 only and Base58 with checksum.